File: ica12/hello.cpp
#include <iostream>
#include <mpi.h>
using namespace std;

int main(int argc, char *argv[])
{
    MPI_Init(&argc, &argv);
    // Get rank/size
    int myRank, numRanks;

    MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
    MPI_Comm_size(MPI_COMM_WORLD, &numRanks);

    if (myRank == 0) cout << "Hello, World!" << endl;

    MPI_Finalize();

    return 0;
}

File: ica12/hello.f90
program hello
    implicit none
    include "mpif.h"

    integer :: ierr, myRank, numRanks

    call MPI_INIT(ierr)
    call MPI_COMM_RANK(MPI_COMM_WORLD, myRank, ierr)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, numRanks, ierr)      

    if (myRank == 0) print *, "Hello World!"

    call MPI_FINALIZE(ierr)
end program hello